apps/leo_storage/src/leo_storage_handler_del_directory.erl

```diff
@@ -0,0 +1,634 @@
+                       ?del_dir_id_list(),
+                       lists:foldl(
+                         fun({_, DelBucketStateBin}, Acc) ->
+                                 case catch binary_to_term(DelBucketStateBin) of
+                                     #del_dir_state{mq_id = MQId,
+                                                    state = ProcState} when ProcState == ?STATE_ONGOING;
+                                                                            ProcState == ?STATE_MONITORING ->
+                                         [MQId|Acc];
+                                     _ ->
+                                         Acc
+                                 end
+                         end, [], DelBucketStateList))),
+            case List of
+                [H|_] ->
+                    Ret = dequeue_1(DelBucketStateList, H),
+                    {Ret, State};
+                [] ->
+                    {ok, State}
+            end;
+        not_found ->
+            {ok, State};
+        {_, Cause} ->
+            ?error("dequeue/1", [{cause, Cause}]),
+            {{error, Cause}, State}
+    end.
+
+%% @private
+-spec(dequeue_1(DelBucketStateList, MQId) ->
+             ok |
+             not_found |
+             {error, Cause} when DelBucketStateList::[],
+                                 MQId::mq_id(),
+                                 Cause::any()).
+dequeue_1([],_) ->
+    not_found;
+dequeue_1([{_, DelBucketStateBin}|DelBucketStateList], MQId) ->
+    case catch binary_to_term(DelBucketStateBin) of
+        #del_dir_state{type = Type,
+                       directory = Directory,
+                       state = ?STATE_PENDING} ->
+            From = self(),
+            _Pid = spawn(
+                     fun() ->
+                             insert_messages(From, MQId, Type, Directory)
+                     end),
+            ok;
+        #del_dir_state{} ->
+            dequeue_1(DelBucketStateList, MQId);
+        {_, Cause} ->
+            ?error("dequeue_1/2", [{cause, Cause}]),
+            {error, Cause}
+    end.
+
+
+%% @private
+-spec(insert_messages(From, MQId, Type, Directory) ->
+             ok | {error, Cause} when From::pid(),
+                                      MQId::mq_id(),
+                                      Type::del_dir_type(),
+                                      Directory::binary(),
+                                      Cause::any()).
+insert_messages(From, MQId, Type, Directory) ->
+    erlang:send(From, {ongoing, MQId, Type, Directory}),
+
+    case leo_storage_handler_object:prefix_search_and_remove_objects(
+           MQId, Directory) of
+        {ok,_TotalMsgs} ->
+            erlang:send(From, {enqueued, MQId, Type, Directory});
+        {error, Cause} ->
+            ?error("insert_messages/3", [{cause, Cause}]),
+            erlang:send(From, {failed, MQId, Type, Directory})
+    end.
+
+
+%% @doc Retrieve available wokers (not ongoingg and monitoring workers)
```

Comment:
The comment (not ongoing and monitoring workers) is different from what actually implemented in the function (pending/ongoing/monitoring state records are returned).